apps/src/weblab/WebLab.js

```diff
@@ -377,7 +380,11 @@ WebLab.prototype.setBrambleHost = function (obj) {
   });
   this.brambleHost.onProjectChanged(this.onProjectChanged.bind(this));
   this.brambleHost.onInspectorChanged(this.onInspectorChanged.bind(this));
-  return project.getCurrentId();
+  if (this.suppliedFilesVersionId) {
+    return `${project.getCurrentId()}-${this.suppliedFilesVersionId}`;
+  } else {
+    return project.getCurrentId();
```

Comment:
Does this have any impact on existing projects? Do we need to migrate anything? My impression reading through the diff is no, but I'd like to get your take.

- I know the set of version ids we're looking at to generate version history for a given project should be different (since we're now looking at the manifest instead of the sources file) but that's probably a nonissue since it wasn't functional before.
- Are we generating versions of the files manifest at a different rate than we were for sources?  In sources we intentionally limit new versions to one per session, so that projects don't rapidly build up a history of hundreds of versions.

Reply:
No impact or migration needed. Fortunately, this path is local to the browser and used with the IndexedDB file system. This change simply makes it possible to view other versions of the same project in a separate tab without running into the problems we usually hit when the same weblab project is opened in multiple tabs simultaneously.

The version ids we're looking at are consistent with the ones we used previously for restoring previous versions of the project (which were files API files-version). But, previously the viewing of a project version didn't work - because it assumed a sources API version.

We are replacing the files manifest using the same logic as before. While working within a "session", we keep replacing the manifest file's latest version with the new one. If you refresh the browser, a new "session" begins, so future changes will result in a new version in version history. This is like other projects. Note that we aren't as aggressive about replacing the individual file versions within a project. These don't appear in version history, so the need to do so for UI purposes is not obvious.

Reply:
Perfect.  I'd forgotten which bug exactly we were fixing here - thanks for the reminder.  This looks great then.